src/test/java/com/commercetools/project/sync/inventoryentry/InventoryEntrySyncerTest.java

```diff
@@ -20,111 +51,130 @@ void setup() {
-  //    when(sourceClient.execute(any(InventoryEntryQuery.class)))
-  //        .thenReturn(CompletableFuture.completedFuture(pagedQueryResult));
-  //
-  //    mockResourceIdsGraphQlRequest(
-  //        sourceClient, "4db98ea6-38dc-4ccb-b20f-466e1566567h", "customTypeKey");
-  //    mockResourceIdsGraphQlRequest(
-  //        sourceClient, "1489488b-f737-4a9e-ba49-2d42d84c4c6f", "channelKey");
-  //
-  //    // test
-  //    final InventoryEntrySyncer inventoryEntrySyncer =
-  //        InventoryEntrySyncer.of(sourceClient, targetClient, mock(Clock.class));
-  //    inventoryEntrySyncer.sync(null, true).toCompletableFuture().join();
-  //
-  //    // assertion
-  //    final LoggingEvent errorLog = syncerTestLogger.getAllLoggingEvents().get(1);
-  //    assertThat(errorLog.getMessage())
-  //        .isEqualTo(
-  //            "Error when trying to sync inventory entry. Existing key: <<not present>>. Update
-  // actions: []");
-  //    assertThat(errorLog.getThrowable().get().getMessage())
-  //        .isEqualTo(
-  //            "InventoryEntryDraft doesn't have a SKU. Please make sure all inventory entry drafts
-  // have SKUs.");
-  //  }
+  @Test
+  void of_ShouldCreateInventoryEntrySyncerInstance() {
+    // test
+    final InventoryEntrySyncer inventorySyncer =
+        InventoryEntrySyncer.of(
+            mock(ProjectApiRoot.class), mock(ProjectApiRoot.class), getMockedClock());
+
+    // assertions
+    assertThat(inventorySyncer).isNotNull();
+    assertThat(inventorySyncer.getSync()).isInstanceOf(InventorySync.class);
+  }
+
+  @Test
+  void transform_ShouldReplaceInventoryEntryReferenceIdsWithKeys() throws JsonProcessingException {
+    // preparation
+    final ProjectApiRoot sourceClient = mock(ProjectApiRoot.class);
+    final InventoryEntrySyncer inventoryEntrySyncer =
+        InventoryEntrySyncer.of(sourceClient, mock(ProjectApiRoot.class), getMockedClock());
+    final List<InventoryEntry> inventoryPage =
+        asList(
+            readObjectFromResource("inventory-sku-1.json", InventoryEntry.class),
+            readObjectFromResource("inventory-sku-2.json", InventoryEntry.class));
+    final List<String> referenceIds =
+        inventoryPage.stream()
+            .filter(inventoryEntry -> inventoryEntry.getSupplyChannel() != null)
+            .filter(inventoryEntry -> inventoryEntry.getCustom() != null)
+            .flatMap(
+                inventoryEntry ->
+                    Stream.of(
+                        inventoryEntry.getCustom().getType().getId(),
+                        inventoryEntry.getSupplyChannel().getId()))
+            .collect(Collectors.toList());
+
+    final String jsonStringCustomTypes =
+        "{\"data\":{\"typeDefinitions\":{\"results\":[{\"id\":\"02e915e7-7763-48d1-83bd-d4e940a1a368\","
+            + "\"key\":\"test-custom-type-key\"}]}}}";
+    final GraphQLResponse customTypesResult =
+        readObject(jsonStringCustomTypes, GraphQLResponse.class);
```

Comment:
```suggestion
        mockResourceIdsGraphQlRequest(sourceClient, "typeDefinitions", "02e915e7-7763-48d1-83bd-d4e940a1a368", "test-custom-type-key");
```

Reply:
I cannot do it here as these 2 mocks are then later used as following:
```
when(graphQLResponse.getBody()).thenReturn(customTypesResult).thenReturn(supplyChannelsResult);
```

When calling graphql endpoint, It returns first the customType and then on the second call it returns supplyChannels.

Reply:
I see. Maybe it's a bit easier to read when mocking the client like here: https://github.com/commercetools/commercetools-project-sync/blob/0b1a239f3e6a83b999a3312c0f6d8ef22c138b50/src/test/java/com/commercetools/project/sync/SyncerFactoryTest.java#L476